### PR TITLE
Fix #44 Annotation Header

### LINKED ISSUE #44: Set lines above initial in the gabc file.
The `gabc` file should be able to set the two lines above the initial.

I have been looking into adding this feature. Currently there can be two `annotation: :` fields in the `gabc` header but they are associated with `voice_info`. Rather than hijack these, perhaps rename `mode: ;` to `modelineone: ;` and add `modelinetwo: ;` For `\gresetfirstlineaboveinitial` and `\gresetsecondlineaboveinitial` respectively.

What is the intended/future use for `annotation: ;`?

### PR DESCRIPTION
**Pull request by eschwab**
This is my first real attempt at C. Review and feedback is most welcome.

Addresses #44

### CODE REVIEW

File: src/struct.h [hunk 1]
```diff
@@ -608,6 +608,8 @@ typedef struct source_info {
  * 
  */
 
+#define NUM_ANNOTATIONS 2
```
Comment (henryso) [inline src/struct.h]: I would rename this `MAX_ANNOTATIONS`.

File: src/gabc/gabc-write.c [hunk 2]
```diff
@@ -820,6 +813,12 @@ void gabc_write_score(FILE *f, gregorio_score *score)
     if (score->mode) {
         fprintf(f, "mode: %d;\n", score->mode);
     }
+    for (annotation_num = 0; annotation_num < NUM_ANNOTATIONS; ++annotation_num) {
+      if (score->annotation[annotation_num]) {
```
Comment (henryso) [inline src/gabc/gabc-write.c]: Should be indented 8 spaces, not 6.

File: src/gabc/gabc-write.c [hunk 3]
```diff
@@ -820,6 +813,12 @@ void gabc_write_score(FILE *f, gregorio_score *score)
...
+	fprintf(f, "annotation: %s;\n",
```
Comment (henryso) [inline src/gabc/gabc-write.c]: I think this should be 12 spaces in, no tab.

**Comment by henryso** [comment]
Stylistically, I think we are using 4-space indentation, no tabs, for C.

**Comment by henryso** [comment]
Question: Is it possible to override the annotations in the gabc file from the tex file?

**Comment by eschwab** [comment]
I think I found all the tabs, I did not have my editor configured properly.

Yes, by default the tex file overrides any annotations in the gabc. The hierarchy is:

 1 `\greannotation` in the main tex file
 2 `annotation` field in gabc file
 3 `mode` field in gabc file

If a higher level is already defined, then the lower ones are ignored.

**Comment by eschwab** [comment]
This is ready for review.

File: src/struct.c [hunk 4]
```diff
@@ -1037,6 +1042,11 @@ static void gregorio_free_score_infos(gregorio_score *score)
     if (score->user_notes) {
         free(score->user_notes);
     }
+    for (annotation_num = 0; annotation_num < MAX_ANNOTATIONS; ++annotation_num) {
```
Comment (henryso) [inline src/struct.c]: Indentation appears to still be 2 spaces here.

**Comment by henryso** [comment]
Besides my indentation comment, this looks fine to me.  Tests check out as well.

**Comment by henryso** [comment]
Looks good. Merging.

**Pull request merged**